### ISSUE TITLE
Add Tensorboard Display of Computational Graph

### ISSUE THREAD

**Issue by dbl001** [karpathy/nanoGPT#390]
I tried to add a Tensorboard SummaryWriter to display the computational graph of nanoGPT.
Unfortunately, because the model's GPT forward method returns logits and loss (instead of just logits), the
SummaryWriter add_graph function gets errors.

I tried using a model wrapper defining the forward function to return only logits.
It works, but the Tensorboard Graph doesn't see what's inside the GPT model.
```
import torch.nn as nn
writer = SummaryWriter()
# Trace the computational graph of the model.
#scripted_model = torch.jit.script(model)
# Add the scripted model to the TensorBoard writer.
#writer.add_graph(scripted_model)
# Extract only tensor outputs
class ModelWrapper(nn.Module):
    def __init__(self, original_model):
        super().__init__()
        self.original_model = original_model

    def forward(self, X):
        logits = self.original_model(X)[0]
        return logits

# Create an instance of the wrapper model
model_wrapper = ModelWrapper(model)

# Use the wrapper model with TensorBoard
writer.add_graph(model_wrapper, X)

```
<img width="186" alt="Screenshot 2023-11-13 at 9 51 12 AM" src="https://github.com/karpathy/nanoGPT/assets/3105499/17d7dbe2-7854-4404-8b9a-36d99b97534e">

Any ideas on how to get Tensorboard to display nanoGPT's computational graph?

**Comment by papertradr** [comment]
click on model wrapper block on tensorboard to see the details of the computational graph

**Comment by dbl001** [comment]
It's working. Thanks!
